pipeline/pipeline_stage.py

```diff
@@ -44,3 +45,8 @@ def start(self):
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def _post_process(self):
+        """
+        Perform any needed cleanup for the stage
+        """
```

Comment:
You need to raise `NotImplementedError` or add a `pass` here. I don't think you can have an empty function in python

Reply:
Mine worked even without this, but you're right we should have this for clarity and consistency